src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -482,6 +482,11 @@ public ProcessStartInfo StartInfo
             {
                 if (_startInfo == null)
                 {
+                    if (Associated && Process.GetCurrentProcessId() != Id)
```

Comment:
Yes Associated returns true when process is started and it does not matter how it was started. But that what I'm expecting. In your example it will not throw because _startInfo is not null. It was assigned on previous steps. And there no way to start process from code and not to assign this variable. 
Anyway I'm going to remove this checking and throw exception even for current process.
```csharp
if (_startInfo == null)
{
   if (Associated && Process.GetCurrentProcessId() != Id)
   {
      throw new InvalidOperationException(SR.CantGetProcessStartInfo);
   }
   _startInfo = new ProcessStartInfo();
}
```